packages/lesswrong/components/posts/PostsByVote.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { useMulti } from '../../lib/crud/withMulti';
+import { registerComponent, Components } from '../../lib/vulcan-lib';
+import { useCurrentUser } from '../common/withUser';
+
+const REVIEW_YEAR = 2020
+
+const PostsByVote = () => {
+    const { SectionTitle, PostsItem2, SingleColumnSection } = Components
+    const currentUser = useCurrentUser()
+    if (!currentUser) return null
+
+    const before = `${REVIEW_YEAR + 1}-01-01`
+    const after = `${REVIEW_YEAR}-01-01`
+
+    const { results: votes } = useMulti({
+        terms: {
+          view: "userPostVotes",
+          userId: currentUser._id,
+          collectionName: "Posts",
+          voteType: "bigUpvote",
+          before: before,
+          after: after
+        },
+        collectionName: "Votes",
+        fragmentName: "UserVotes",
+        limit: 1000
+      });
+
+    if (!votes) return null
+    const postIds = votes.map(vote=>vote.documentId)
+
+    const { results: posts } = useMulti({
```

Comment:
Lint errors are real (due to the `currentUser` null check being before hooks). Either split this into two components the inner of which takes a nonnull user, or move the check down and put a `skip` on the query hooks.